### DISCUSSION TITLE
LANraragi v.0.8.0 - Black Tie White Noise

### DISCUSSION THREAD

**Discussion by Difegue** [Difegue/LANraragi#509] [Show and tell]
![d886a1488d50c54fb2111f45f7f9be3d29daca8e45104468c03c129e590a39d5](https://user-images.githubusercontent.com/8237712/134776616-7eb068fb-061a-4ff2-b2eb-6e3831c21ec6.png)

Hello! It's been a little while (compared to the previous release schedule anyway), but I hope this new release finds you well.  

This version brings mostly small improvements and polish, although *Tag Rules* are a super cool new feature I think some people will enjoy.  
I wanted to hold it back a bit longer to add at least one more feature, but the Fakku plugin was broken and shit now so uhhh here catch

## Release Notes are now shown in the Index once upon updating

If you've never read LRR patchnotes before, welcome!   

Now you'll have to. 🔫    

## The Blacklist feature has been removed and replaced with Tag Rules! (#497)

![image](https://user-images.githubusercontent.com/8237712/134780271-58446531-f866-446a-b63e-971c61f1d80a.png)

Tag Rules do everything the blacklist used to do and _much more_, so please check it out! 🥳  
The full documentation is available here: https://sugoi.gitbook.io/lanraragi/advanced-usage/tag-rules
I'd like to thank @IceBreeze once again for contributing this improvement.  

If you had custom blacklist tags, you can migrate them to Tag Rules using the migration script available in **Plugins**:  
![image](https://user-images.githubusercontent.com/8237712/134780351-f40ef3e4-663f-4f61-8134-b0dcc431d263.png)  

## New Tag Editing UI

![image](https://user-images.githubusercontent.com/8237712/134779970-805733e2-4c84-49be-899b-94eeb2b1fc03.png)
Tag editing now displays your archive's tags as a neat list of items, and will make suggestions using the stats API. 🎊 

## Other Changes

* ### Snapshot to RDB less often on Docker/Windows and start enabling AOF 
  💿 Moving Minion jobs to Redis meant the old method of writing to disk as often as possible led to a lot of unnecessary hard drive work.
  📀 I've slowed down the amount of writes a lot (essentially moving back to the default), and will fully migrate LRR to use [AOF](https://redis.io/topics/persistence) with RDB backups as the database storage in future releases. 
  💽 For now, you shouldn't see any changes besides a new file next to `database.rdb`.

* ### (Small) Search engine optimizations
  🚗 I've made a number of adjustments to the search model to reduce the amount of Redis calls. This improves search speed __a bit__, but there's still improvements to be made here.

* ### Windows Installer enhancements
  🐱‍🐉 The WSL Distro will now be installed in the installation folder of the bootstrapper you define, instead of defaulting to AppData.
  🐱‍👤 The Windows installer no longer relies on LxRunOffline, instead using `wsl.exe` directly for everything. 
  🖥️ The `cmd` windows that install/uninstall the distro now autoclose as a result, making for a more pleasant install experience.
  💀 This means support for Windows 10 1809 and below has been **dropped**. aye im sorry m8

* ### Nightlies have moved off MEGA
  🌑 You can now find the latest nightly build here: https://nightly.link/Difegue/LANraragi/workflows/push-continous-delivery/dev/LANraragi.msi 
  🗑️ While this means older nightlies aren't easily accessible anymore, it also means I don't have to empty the MEGA cloud drive manually every 3 months or so. ¯\\\_(ツ)_/¯

* ### Metadata plugin fixes
  🌮 Update chaika plugin to match new API result syntax
  🍔 Fix EH plugin not using the UserAgent object properly
  🍟 FAKKU plugin now supports cookies to access __questionable__ content (Thanks @Nodja !)
  🌭 (#504) Update FAKKU Plugin to match the site redesign 
  🥓 (#484) Switch EH API url, and use useragent everywhere 

* ### API Changes
  🔑 Add a minweight parameter to `/api/db/stats` to filter out tags that don't have enough occurences in the DB 
  🔑 Fix metadata edition API not accepting empty strings
  🔑  (#491) Fix UTF8 character encoding in `/api/database/backup` and `/api/database/stats`

* ### Miscellaneous bugfixes and improvements
  💎 (#495) Replace logical ORs as they're still too fresh for some browsers (mostly mobile)
  💎 Force Linux::Inotify 2.2 to sidestep a bug on WSL (`cannot open fd 5 as perl handle`)
  💎 (https://github.com/Difegue/Tsukihi/issues/5) Normalize e-h and ex links in SourceFinder
  💎 Fix GS command for PDFs with unicode characters
  💎 (#466) Clear out tooltips on datatables drawcallback 
  💎 Various CSS touchups, including better support for small screens in the index
  💎 Add current search terms to window title
  💎 Fix search engine potentially returning null entries in the JSON
  💎 Update Datatables to 1.10.25 and Mojolicious to 9.19
  💎 Support CORS preflight requests if the CORS option is enabled in settings
  💎 (#450) `source:` tag divs now lead to the source URL instead of doing a search
# 

_The very JG Ballard image of a rich man standing in his hotel suite, watching a riot unfold in the city below and feeling vaguely euphoric about it, would seem ripe inspiration for someone who’d once written “Panic In Detroit.” Instead, Bowie’s L.A. riots song was “Black Tie White Noise,” a track teetering between dark sarcasm and watery humanism._

<hr /><em>This discussion was created from the release <a href='https://github.com/Difegue/LANraragi/releases/tag/v.0.8.0'>LANraragi v.0.8.0 - Black Tie White Noise</a>.</em>

**Comment by Cordisl** [comment]
What to enter in api and source. App name dumanga![image](https://user-images.githubusercontent.com/91519164/135028354-323da6df-120d-4b81-9d3b-142acde46ae0.jpeg)

**Reply by Difegue** [reply]
You need to type in the IP address of the computer running the server. 
As for the API key, it has to be set by you in the server settings.

Check the documentation for details 👍

**Comment by IceBreeze** [comment]
I think this closes #351

**Reply by Difegue** [reply]
Correct, I somehow skipped over this issue when triaging before the release. Thanks!
